Source/WebCore/Modules/webcodecs/WebCodecsBase.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebCodecsBase.h"
+
+#if ENABLE(WEB_CODECS)
+
+#include "Event.h"
+#include "EventNames.h"
+#include "WebCodecsControlMessage.h"
+
+#include <wtf/TZoneMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_OR_ISO_ALLOCATED_IMPL(WebCodecsBase);
+
+WebCodecsBase::WebCodecsBase(ScriptExecutionContext& context)
+    : ActiveDOMObject(&context)
+{
+}
+
+WebCodecsBase::~WebCodecsBase() = default;
+
+void WebCodecsBase::queueControlMessageAndProcess(WebCodecsControlMessage&& message)
+{
+    if (m_isMessageQueueBlocked) {
+        m_controlMessageQueue.append(WTFMove(message));
+        return;
+    }
+    m_controlMessageQueue.append(WTFMove(message));
+    processControlMessageQueue();
+}
+
+void WebCodecsBase::queueControlMessageForCodecOperationAndProcess(WebCodecsControlMessage&& message)
+{
+    incrementOperationQueueSize();
```

Comment:
This is not as per spec, some messages will increment the queue (decode) but not all of them (configure).

Reply:
That's right for decode/encode operations you call `queueControlMessageForCodecOperationAndProcess` otherwise you call `queueControlMessageAndProcess`

Reply:
`queueControlMessageForCodecOperationAndProcess` is only called for the decode or encode operation, the other messages will not increase the "OperationQueueSize" maybe I should rename it to `incrementCodecOperationQueueSize()` and `decrementCodecOperationQueueSize();